src/main/java/com/theelixrlabs/healthcare/service/DoctorPatientAssignmentService.java

```diff
@@ -48,6 +48,34 @@ public DoctorPatientAssignmentDto assignDoctorToPatient(DoctorPatientAssignmentD
         return mapDoctorPatientAssignment(doctorPatientAssignmentDto);
     }
 
+    /**
+     * checks for doctor patient assignment and if present then unassigns doctor from patient
+     *
+     * @param doctorPatientAssignmentDto DTO object containing doctorId and patientId
+     * @throws CustomException if any exception occurs
+     */
+    public void unassignDoctorFromPatient(DoctorPatientAssignmentDto doctorPatientAssignmentDto) throws CustomException {
+        validateDoctorPatientAssignment(doctorPatientAssignmentDto);
+        UUID doctorId = UUID.fromString(doctorPatientAssignmentDto.getDoctorId());
+        UUID patientId = UUID.fromString(doctorPatientAssignmentDto.getPatientId());
+        Optional<DoctorPatientAssignmentModel> optionalDoctorPatientAssignmentModel =
+                doctorPatientAssignmentRepository.findByDoctorIdAndPatientIdAndDateOfUnassignmentNull(doctorId, patientId);
+        if (!doctorPatientAssignmentRepository.existsByDoctorIdAndPatientId(doctorId, patientId)) {
+            throw new CustomException(messageUtil.getMessage(DoctorPatientAssignmentConstants.NO_ASSIGNMENT_EXISTS_KEY));
+        } else if (optionalDoctorPatientAssignmentModel.isEmpty()) {
+            throw new CustomException(messageUtil.getMessage(DoctorPatientAssignmentConstants.DOCTOR_ALREADY_UNASSIGNED_KEY));
+        }
```

Comment:
Do you actually need else if (_)?
why simply if (_) ?

Reply:
Resolved